services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowOperationsService.java

```diff
@@ -279,9 +288,43 @@ public UnidirectionalFlow updateFlow(FlowDto flow) throws FlowNotFoundException
 
             flowRepository.createOrUpdate(currentFlow);
 
-            return Optional.of(forwardFlow);
+            return Optional.of(result.updatedFlow(forwardFlow).build());
 
         }).orElseThrow(() -> new FlowNotFoundException(flow.getFlowId()));
+
+        if (updateFlowResult.isNeedRerouteFlow()) {
+            Flow updatedFlow = updateFlowResult.getUpdatedFlow().getFlow();
+            Set<PathId> pathIds = new HashSet<>(updatedFlow.getFlowPathIds());
```

Comment:
I'm not sure about pathIds. I guess we must get pathIds for reroute from original flow.

Reply:
This flow update does not affect the flow paths at this point.